topography/topography.F90

```diff
@@ -224,651 +253,115 @@ module topography_mod
- !! if the Navy 1/6 degree percent water data set was not readable.
- function get_water_mask_1d (blon, blat, water_mask)
-
- real, intent(in),  dimension(:)   :: blon !< The longitude (in radians) at grid box boundaries.
- real, intent(in),  dimension(:)   :: blat !< The latitude (in radians) at grid box boundaries.
- logical, intent(out), dimension(:,:) :: water_mask !< A binary mask for water (true) or land (false).
-                                     !! The size of this field must be size(blon)-1 by size(blat)-1.
- logical :: get_water_mask_1d
-
- real, dimension(size(water_mask,1),size(water_mask,2)) :: water_frac
-!-----------------------------------------------------------------------
-   if (.not. module_is_initialized) call topography_init()
-
-   if ( get_water_frac_1d(blon, blat, water_frac) ) then
-     where (water_frac > 0.50)
-       water_mask = .true.
-     elsewhere
-       water_mask = .false.
-     end where
-     get_water_mask_1d = .true.
-   else
-     get_water_mask_1d = .false.
-   endif
-!-----------------------------------------------------------------------
-
- end function get_water_mask_1d
+subroutine topography_init ()
+   if ( module_is_initialized ) return
+   call write_version_number("TOPOGRAPHY_MOD", version)
+   call read_namelist
+   module_is_initialized = .TRUE.
+end subroutine topography_init
 
 !#######################################################################
-
- function get_water_mask_2d (blon, blat, water_mask)
-
- real, intent(in),  dimension(:,:)   :: blon !< The longitude (in radians) at grid box boundaries.
- real, intent(in),  dimension(:,:)   :: blat !< The latitude (in radians) at grid box boundaries.
- logical, intent(out), dimension(:,:) :: water_mask !< A binary mask for water (true) or land (false).
-                                     !! The size of this field must be size(blon)-1 by size(blat)-1.
- logical :: get_water_mask_2d
- real, dimension(size(water_mask,1),size(water_mask,2)) :: water_frac
-!-----------------------------------------------------------------------
-   if (.not. module_is_initialized) call topography_init()
-
-   if ( get_water_frac_2d(blon, blat, water_frac) ) then
-     where (water_frac > 0.50)
-       water_mask = .true.
-     elsewhere
-       water_mask = .false.
-     end where
-     get_water_mask_2d = .true.
-   else
-     get_water_mask_2d = .false.
-   endif
-
-!-----------------------------------------------------------------------
-
- end function get_water_mask_2d
-
 !#######################################################################
 !##################   private interfaces below here   ##################
 !#######################################################################
 
- function open_topog_file ( )
- logical :: open_topog_file
- real    :: r_ipts, r_jpts
- integer :: namelen
+function open_topog_file ( )
+logical :: open_topog_file
+real(kind=r8_kind)    :: r_ipts, r_jpts
```

Comment:
Im wondering if these could be r4_kind if the whole purpose of this subroutine is to open the topog_file and get ipts and jpts